### ISSUE TITLE
Uncaught RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by ljosberinn** [welldone-software/why-did-you-render#15]
On this [commit](https://github.com/ljosberinn/resources-helper/commit/aa5233e7ff428452a8fcf216d28d00fc39f45fa1), as soon as I use even only try to start my app with `/registration`, it gets into an infinite loop. Edit: the other, technically working, routes `/login`, `/factories` and `/mines` also crash.

```
The above error occurred in the <Registration> component:
    in Registration (created by Context.Consumer)
    in Route (at Routes.tsx:28)
    in Switch (at Routes.tsx:19)
    in main (at Routes.tsx:18)
    in Routes (at RHelper.tsx:17)
    in RHelper (at src/index.tsx:30)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Context.Consumer)
    in ConnectedRouterWithContext (created by Context.Consumer)
    in Connect(ConnectedRouterWithContext) (at src/index.tsx:29)
    in PersistGate (at src/index.tsx:28)
    in Provider (at src/index.tsx:27)

```

It's stuck on [`useState(false)`](https://github.com/ljosberinn/resources-helper/blob/rhelper4/src/components/Authentication/Registration.tsx#L18)
![Bildschirmfoto 2019-04-05 um 20 06 17](https://user-images.githubusercontent.com/29307652/55647585-50b66e00-57de-11e9-8d5b-cae956fc73c5.png)

one level up:
![Bildschirmfoto 2019-04-05 um 20 07 24](https://user-images.githubusercontent.com/29307652/55647640-75aae100-57de-11e9-861d-34340186e828.png)

Did I miss anything concerning documentation or is it the mix of tools I use?

**Comment by vzaidman** [comment]
checking

**Comment by vzaidman** [comment]
fixed in version 3.0.2
